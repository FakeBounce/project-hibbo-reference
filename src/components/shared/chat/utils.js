import dateFormat from 'dateformat';

const DEPRECATION_MESSAGE =
  'isSameUser and isSameDay should be imported from the utils module instead of using the props functions';

export const isSameDay = (currentMessage = {}, diffMessage = {}) => {
  if (!currentMessage.createdAt || !currentMessage.createdAt) {
    return false;
  }
  try {
    const currentCreatedAt = dateFormat(currentMessage.createdAt, 'd/m/yyyy');
    const diffCreatedAt = dateFormat(diffMessage.createdAt, 'd/m/yyyy');
    return currentCreatedAt === diffCreatedAt;
  } catch (err) {
    return false;
  }
};

export const isSameUser = (currentMessage = {}, diffMessage = {}) => {
  return Boolean(
    diffMessage.user &&
      currentMessage.user &&
      diffMessage.user._id === currentMessage.user._id,
  );
};

export const warnDeprecated = fn => {
  return (...args) => {
    console.warn(DEPRECATION_MESSAGE); // eslint-disable-line
    return fn(...args);
  };
};
