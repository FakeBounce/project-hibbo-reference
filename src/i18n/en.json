{"105202":"Card number is not valid","105203":"Expiry date is not valid","105204":"Cryptogram is not valid","accesspiggy.button.askAccess":"Demander l'acces à une tirelire","accesspiggy.button.byId":"Je possède l'ID","account.delete.success":"Your account has been deleted","account.error.unauthorized":"You are not authorized to access this information","account.password":"Your password has been reset","account.password.reset.success":"Reset of password success","account.verify.page":"Thank you for checking your account","auth.label.address":"Address","auth.label.birthday":"Date of birth","auth.label.city":"City","auth.label.confirmpassword":"Confirm","auth.label.country":"Country","auth.label.createAccount":"I don't have an account yet","auth.label.email":"Email","auth.label.entermailcode":"Enter the code sent by mail to test your password","auth.label.firstname":"First name","auth.label.forgottenPwd":"You forgot your password?","auth.label.hidepassword":"Hide the password","auth.label.lastname":"Last name","auth.label.minimunpasswordsize":"6 numbers minimun","auth.label.newpassword":"New password","auth.label.password":"Password","auth.label.phone":"Phone number","auth.label.postalcode":"Zip code","auth.label.region":"Region","auth.label.securityCode":"Security code","auth.label.showpassword":"show the password","auth.label.ssn":"Last 4 digits of your SSN","auth.label.state":"State","auth.login":"Login","auth.provider.fb":"Facebook","auth.provider.google":"Google","auth.provider.mail":"Email","auth.register":"Register","auth.security.passwordNotStrong":"your password to medium security","auth.security.passwordStrong":"your password to strong security","auth.security.passwordWeak":"your password to low security","auth.terms.invalidDate":"Invalide date","auth.terms.link":"https://stripe.com/legal","auth.terms.privacy":"I accept your privacy policy","auth.terms.service":"I accept your terms of service","auth.terms.validateTerms":"Terms must be validated","cardScanner.crypto":"Cryptogram","cardScanner.expiracy":"Expire at","cardScanner.expire":"EXPIRE","cardScanner.expire.end":"AT END","cardScanner.instruction":"Center your card inside the brackets","cardScanner.label.button":"Save","cardScanner.label.cardHolder":"Card holder","cardScanner.label.cardNumber":"Card number","cardScanner.label.crypto":"CVV","cardScanner.label.currency":"Currency","cardScanner.label.expiracyMonth":"MM","cardScanner.label.expiracyYear":"YY","cardScanner.title":"Enter your card info","cardScanner.titulary":"Card holder","cardScanner.visual.cardHolder":"CARD HOLDER","cardScanner.visual.cardNumber":"CARD NUMBER","cardScanner.visual.cvv":"CVV","cardScanner.visual.date":"DATE","cardScanner.visual.month":"MONTH","child.ad.comingsoon":"Coming soon","child.ad.crystalegg":"The Crystal egg","child.ad.episode":"Learn addictions Episode I","child.ad.newgame":"New game is coming soon","code.header.step0":"Security code","code.header.step1":"Confirmation","code.label.step0":"Create your security code","code.label.step1":"Confirm your security code","code.pay.error":"This is not the right code","code.setup.error":"Codes don't match","common.form.address":"Address","common.form.cancel":"Cancel","common.form.city":"City","common.form.email":"Email","common.form.firstname":"First name","common.form.lastname":"Last name","common.form.phone":"Phone number","common.form.postal":"Zip","common.form.save":"Save","common.form.state":"State","contri.pairing.addPiggy":"Add a Monimalz","contri.pairing.addPiggyById":"I want to access to the Monimalz of","contri.pairing.askAccess":"Ask access for the Monimalz","contri.pairing.howToAddaPiggy":"How can I add the Monimalz?","email.confirmation.content":"You are officially part of the family of Monimalz' breeders !\n<br />See you soon for new adventures","email.confirmation.header":"Welcome  /%firstname%/ !<br />\nYou just adopted a nice /%piggy%/","email.confirmation.subject":"Registration confirmed!","email.passwordChange.content":"Your new password has been successfully saved","email.passwordChange.header":"Yes! It's now done!<br />Tadammm !","email.passwordChange.subject":"Password changed","email.passwordReset.content":"A reset of your password on your account <br />\nhas been requested on the Monimalz app. <br />\nClick or copy the link below and complete the rest of the form!<br /> <br />","email.passwordReset.header":"Oh no!<br />Did you really lost your password?","email.passwordReset.subject":"Password forgotten","email.shopConfirmation.content":"Thanks, your order n° /%packageNumber%/<br />\nhas been taken into account","email.shopConfirmation.header":"Hi sweetie <br />\nYour order is on its way ","email.shopConfirmation.subject":"Order confirmation n° /%packageNumber%/","email.verification.content":"To finalize your registration,<br />\nconfirm your email address<br />\nby clicking on the link below","email.verification.header":"One last-ditch effort!","email.verification.subject":"Confirmation of email","error.account.alreadyexists":"This user already exists","error.account.expiredToken":"Expired token","error.account.invalidToken":"Token not valid","error.account.notfound":"User not found","error.account.password":"Wrong password","error.account.providerIdMismatch":"Provider not recognized. Use your Facebook, Google or email account.","error.account.reset.max.try":"","error.account.reset.wrong.token":"Wrong token, you still have /%availableTry%/ tries","error.apikey.match":"Your application is outdated","error.auth.fb":"Facebook has thrown an error","error.auth.google":"Google has thrown an error","error.auth.google.cancel":"You canceled the connection to Google","error.bank.notexists":"Your bank account is not valid","error.card.notexists":"Your card is not valid","error.email.valid":"This e-mail address is not valid","error.eshopItem.notexists":"This item is not available","error.generic":"An unexpected error happened","error.mission.notexists":"This mission does not exist anymore","error.moneypot.notexists":"Moneypot is not available yet","error.piggy.alreadyexists":"This money box is already registered ","error.piggy.notexists":"this money box does not exist","error.profile.notexists":"This profile does not exist","error.profile.notexists.stripe":"This profile stripe does not exist","error.server.not.connected":"Impossible de se connecter au serveur","error.transaction.failed":"/%resultCode%/ : /%resultMessage%/","error.transaction.notexists":"This transaction does not exist","error.validation.accountId":"This account cannot be validated","error.validation.amount":"","error.validation.apiKey":"","error.validation.avatar":"","error.validation.bdate":"","error.validation.billingAddress":"","error.validation.email":"This mail address cannot be validated","error.validation.equipments":"","error.validation.eshop.amount":"","error.validation.eshop.buyerProfileId":"","error.validation.eshop.date":"","error.validation.eshop.description":"","error.validation.eshop.icon":"","error.validation.eshop.itemsPurchase":"","error.validation.eshop.name":"","error.validation.eshop.price":"","error.validation.eshop.recieverProfileId":"","error.validation.eshop.status":"","error.validation.eshop.theme":"","error.validation.firstname":"","error.validation.homeAddress":"","error.validation.lastname":"","error.validation.mission.amount":"","error.validation.mission.creationDate":"","error.validation.mission.deadline":"","error.validation.mission.icon":"","error.validation.mission.profileAssigneeId":"","error.validation.mission.profileGiverId":"","error.validation.mission.status":"","error.validation.mission.theme":"","error.validation.mission.title":"","error.validation.moneypot.amount":"","error.validation.moneypot.authorizedProfilesId":"","error.validation.moneypot.creatorProfileId":"","error.validation.moneypot.deadline":"","error.validation.moneypot.slug":"","error.validation.moneypot.theme":"","error.validation.moneypot.title":"","error.validation.moneypot.type":"","error.validation.nickname":"","error.validation.password":"","error.validation.paymentmeansId":"","error.validation.phone":"","error.validation.piggybankId":"","error.validation.profileType":"","error.validation.transaction.amount":"","error.validation.transaction.date":"","error.validation.transaction.icon":"","error.validation.transaction.message":"","error.validation.transaction.profileGiverId":"","error.validation.transaction.profileRecieverId":"","error.validation.transaction.status":"","error.validation.transaction.theme":"","error.validation.transaction.type":"","error.validation.uuid":"","error.wallet.notexists":"","eshopItem.delete.success":"","history.balance.date":"Balance on the","history.bank":"Bank","history.date.from":"on the","history.livret":"Account","history.piggy":"Piggy","history.row.mission":"Mission","history.transfert":"Transfer","messenger.default.message.bubble.credit":"","messenger.default.message.bubble.debit":"","messenger.error.no.message":"You did not fill in your message for /%nickname%/","messenger.input.placeholder":"Message","messenger.main.placeholder":"Write a small note to /%firstname%/","mission.add.amountError":"Amount cannot be empty","mission.add.dateError":"Invalid date","mission.add.iconError":"Icon required","mission.add.label":"Create","mission.add.title":"Add a mission","mission.add.titleError":"Title can not be empty","mission.complete.message":"Mission accomplished !","mission.completed.text":"Mission accomplished !","mission.create.message":"Hey! You just got a new mission from /% nickname% /! Look at it now:% title%H","mission.deadline.expired":"Expired","mission.deadline.text":"D-","mission.delete.success":"Mission successfully deleted","mission.detail.paid":"Paid","mission.detail.pay":"Pay now","mission.detail.relaunch":"Relaunch mission ?","mission.detail.title":"The mission","mission.edit.label":"Save","mission.edit.title":"Edit a mission","mission.label.amount":"2","mission.label.expiracyDay":"dd","mission.label.expiracyMonth":"mm","mission.label.expiracyYear":"yyyy","mission.label.title":"Clean your room","mission.menu.add":"Add +","mission.menu.done":"Cancel","mission.menu.homeTitle":"Mission","mission.menu.modify":"Modify","mission.success.message":"Well done! You have completed your mission /% title% / and you have just received /% amount%/€ from /%nickname%/!","moneypot.delete.success":"Your pool has been removed from your Money Pot!","monimalz":"Monimalz","navigation.goals":"Missions","navigation.messages":"Messages","navigation.moneyPot":"Money Pot","navigation.myMonimalz":"My Monimalz","navigation.piggyconnect":"Tirelire Access","navigation.reminders":"Reminders","navigation.settings":"Settings","navigation.shop":"Shop","navigation.transfert":"Transfer","notification.header.main":"Messages","openMonimalz.button.close":"Close","openMonimalz.button.emptyMoneybox":"Yes, I cleared out my Monimalz","openMonimalz.button.mistake":"Oops, it was a mistake","openMonimalz.button.moneyDeposit":"Yes, I added money inside","openMonimalz.button.sendTo":"Send to","openMonimalz.button.withdrawal":"I withdraw a small amount","openMonimalz.subtile.dropMoney":"How much would you add?","openMonimalz.subtile.opened":"Your Monimalz is open!","openMonimalz.subtile.resetMoney":"Your balance has been reset to 0","openMonimalz.subtile.sameMoney":"Your balance stays the same","openMonimalz.subtile.takeMoney":"How much would you take?","openMonimalz.title.daccord":"OK","openMonimalz.title.hey":"Hey!","pairing.cancel":"Cancel","pairing.creation.error":"Some error happened while adding your piggy. You should try again later ","pairing.day":"dd","pairing.miko":"Miko","pairing.mikoDescr":"le panda tout doux","pairing.momo":"Momo","pairing.momoDescr":"le petit singe fou","pairing.mona":"Mona","pairing.monaDescr":"la baleine rigolote","pairing.month":"mm","pairing.name":"Firstname","pairing.noPic":"Use a picture of a Monimalz","pairing.pwd":"Password","pairing.selectFromPic":"Select from your pictures","pairing.selection.1":"Hello there !","pairing.selection.2":"Which Monimalz","pairing.selection.3":"did you adopt ?","pairing.selection.monkey1":"Hello it's Momo","pairing.selection.monkey2":"Is it me you","pairing.selection.monkey3":"adopted ?","pairing.selection.panda1":"Hello it's Popo","pairing.selection.panda2":"Is it me you","pairing.selection.panda3":"adopted ?","pairing.selection.thank1":"Merci de","pairing.selection.thank2":"m'avoir adopté !","pairing.selection.thank3":"Je suis","pairing.selection.whale1":"Hello it's Toto","pairing.selection.whale2":"Is it me you","pairing.selection.whale3":"adopted ?","pairing.selectPic":"Select a picture","pairing.step1.button":"Connecter","pairing.step2.button":"Validate","pairing.step3.button1":"Demoiselle","pairing.step3.button2":"P'tit gars","pairing.step4.button":"Next","pairing.step6.button":"Je valide la photo","pairing.step8.button1":"Ok","pairing.step8.button2":"Plus tard","pairing.step9.button1":"Type the numbers","pairing.step9.button2":"Scan my card","pairing.takePic":"Take a picture","pairing.title1.label1":"What if we connected","pairing.title1.label2":"between each other","pairing.title1.label3":"What is your wifi ?","pairing.title2.label1":"Quel est le","pairing.title2.label2":"mot de passe ?","pairing.title2.label3":"Chut, je le garde","pairing.title2.label4":"pour moi...","pairing.title3.label1":"Mon compagnon","pairing.title3.label2":"d'adoption sera une","pairing.title3.label3":"demoiselle ou","pairing.title3.label4":"un ptit gars ?","pairing.title4.label1":"What's her ","pairing.title4.label2":"What's his ","pairing.title4.label3":"little name?","pairing.title5.label1":"What does she","pairing.title5.label2":"What does he","pairing.title5.label3":"look like","pairing.title6.label1":"She's cute","pairing.title6.label2":"He's cute","pairing.title7.label1":"Remind me her","pairing.title7.label2":"Remind me his","pairing.title7.label3":"birthday","pairing.title8.label1":"Ajoute une carte","pairing.title8.label2":"et faites un premier","pairing.title8.label3":"versement","pairing.title9.label1":"Add a card","pairing.title9.label2":"for a first","pairing.title9.label3":"payment","pairing.year":"yyyy","piggy.delete.success":"/%firstname%/'s Monimalz has been successfully deleted","privacy.paragraph1":"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do","privacy.title":"Privacy policies","profile.delete.success":"The profile of /%firstname%/ has been successfully deleted","settings.childProfile.accountBalance":"Account balance","settings.childProfile.accountNumber":"Account number","settings.childProfile.birthDay":"Date of birth","settings.childProfile.piggyBankBalance":"Money box balance","settings.childProfile.standBy":"Sleep","settings.childProfile.transfer":"Transfer","settings.parentProfile.address":"Address","settings.parentProfile.profileOfName.consonant":"/%name%/'s profile","settings.parentProfile.profileOfName.vowel":"/%name%/'s profile","settings.purchases.amount":"Amount","settings.purchases.date":"Date","settings.purchases.followStatusOfYourOrderWith":"Follow the status of your order with ","settings.purchases.myOrders":"My orders","settings.purchases.orderId":"Order number","settings.summary.multipleOngoingPurchases":"/%number%/ current orders","settings.summary.myPaymentMeans":"My payment means","settings.summary.myPiggyBanks":"My piggy banks","settings.summary.myProfile":"My profile","settings.summary.myPurchases":"My orders","settings.summary.noOngoingPurchases":"No current orders","settings.summary.oneOngoingPurchase":"One current order","settings.transfer.onTheAccountOfName.consonant":"On /%name%/'s account","settings.transfer.onTheAccountOfName.vowel":"On /%name%/'s account","settings.transfer.piggyBankOfName.consonant":"/%name%/'s Monimalz","settings.transfer.piggyBankOfName.vowel":"/%name%/'s Monimalz","settings.transfer.subTitle":"I want to transfer","shared.no":"no","shared.yes":"yes","terms.intro":"Remplis les informations nécessaires pour faire ton virement et confirme que tu as bien lu les conditions d’utilisation.","terms.title":"Terms of Service","toast.email":"This email adress is not correct","toast.empty":"The imput is empty","toast.notcorrectpasswordsize":"password must be 6 charactere","toast.notsamepassword":"not same password","toast.password":"This password is not correct","toast.phone":"This phone number is not correct","transaction.date":"on the","transaction.delete.success":"This transaction has been successfully canceled","transaction.firstOne":"Do your first payment","transfert.button.sendTo":"Send to","transfert.confirm.card.missing":"You need to choose a card first","transfert.confirm.code.error":"The security code used is not correct","transfert.confirm.fingerPrint.error":"Your fingerprint is not correct, type in your security code instead","transfert.debit.step1.add.button":" Add a recipient","transfert.debit.step1.header":"/%amount%/€ ","transfert.debit.step2.header":"Transfer","transfert.debit.step2.subtitle":"On the account of /%fullName%/","transfert.debit.step3.amountConfirmation.consonant":"Confirm your debit of /%amount%/€ to the account of /%fullName%/","transfert.debit.step3.amountConfirmation.vowel":"Confirm your debit of /%amount%/€ to the account of /%fullName%/","transfert.debit.step3.header":"Confirmation","transfert.header.debit":"Debit","transfert.header.moneyPot":"Money Pot","transfert.header.pay":"Payment","transfert.moneypot.main.first":"Pleasure","transfert.moneypot.main.second":"to offer bigger!","transfert.moneypot.message":"Coming soon","transfert.pay.addCard":"Add a new credit card","transfert.pay.step1.amountConfirmation":"Confirm your payment of /%amount%/€","transfert.pay.step1.header":"Enter your code","transfert.pay.step2.bubble.amountConfirmation.part1":"You've just","transfert.pay.step2.bubble.amountConfirmation.part2":"Tranfered to /%firstname%/","transfert.pay.step2.bubble.chooseAction":"Voulez-vous envoyé associé un message à ce versement ?","transfert.pay.step2.button.message":"A message","transfert.pay.step2.button.mission":"A mission","transfert.touchid.ask":"Use Touch Id or enter your security code","transfert.touchId.popup":"Press your finger","transfert.recipient.accountnumber":"IBAN","transfert.recipient.formtitle":"Saisis manuellement les infos pour envoyer directement l’argent sur son compte bancaire","transfert.recipient.accountholdername":"Titulaire du compte","transfert.recipient.accountnumberplaceholder":"Numéro de IBAN"}
