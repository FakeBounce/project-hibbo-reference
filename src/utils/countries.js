export const contriesList = [
  ['France', 'FR'],
  ['United States', 'US'],
  ['Spain', 'ES'],
  ['Italy', 'IT'],
  ['Germany', 'DE'],
  ['United Kingdom', 'GB'],
];

export const countriesCurrencies = [
  ['Euros', 'EUR'],
  ['Dollar', 'USD'],
  ['Pound', 'GBP'],
];
