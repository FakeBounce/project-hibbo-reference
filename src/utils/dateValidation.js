const daysInMonth = (m, y) => {
  switch (m) {
    case 1:
      return (y % 4 === 0 && y % 100) || y % 400 === 0 ? 29 : 28;
    case 8:
    case 3:
    case 5:
    case 10:
      return 30;
    default:
      return 31;
  }
};

const isValidDate = (d, m, y, futur = false) => {
  const dd = parseInt(d, 10);
  const mm = parseInt(m, 10) - 1;
  const yy = parseInt(y, 10);
  const t = new Date();
  return (
    mm >= 0 &&
    mm < 12 &&
    dd > 0 &&
    dd <= daysInMonth(mm, yy) &&
    (futur ? yy <= t.getFullYear() : yy >= t.getFullYear())
  );
};

export default isValidDate;
