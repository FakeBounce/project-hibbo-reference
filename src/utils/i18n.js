import { Platform, NativeModules } from 'react-native';

import messages from '../i18n';

const startsWithVowel = param => {
  const letter = param[0];
  if (
    letter === 'a' ||
    letter === 'e' ||
    letter === 'i' ||
    letter === 'o' ||
    letter === 'u' ||
    letter === 'y'
  ) {
    return true;
  }

  return false;
};

export const getLanguage = () => {
  if (Platform.OS === 'ios' && NativeModules.SettingsManager) {
    if (
      NativeModules.SettingsManager.settings.AppleLocale.substr(0, 2) === 'fr'
    ) {
      return 'fr';
    }
  } else if (Platform.OS === 'android' && NativeModules.I18nManager) {
    if (NativeModules.I18nManager.localeIdentifier.substr(0, 2) === 'fr') {
      return 'fr';
    }
  }
  return 'en';
};

const getTrad = (target, key, defaultKey) => {
  return target[key] || defaultKey;
};

export const getTranslations = (key, params, elisionParam = null) => {
  const lang = getLanguage();
  let newKey = key;

  // Handle elision of the translation depending on a param
  if (elisionParam && params[elisionParam]) {
    newKey = startsWithVowel(params[elisionParam])
      ? `${key}.vowel`
      : `${key}.consonant`;
  }

  // Proxy doesn't work on old version of IOS and android
  // const i18n = new Proxy(messages[lang], {
  //   get(target, k) {
  //     return target[k] || key;
  //   },
  // });
  const res = getTrad(messages[lang], newKey, key);

  if (!res) return res;
  return res.replace(/\/%(.*?)%\//g, (matching, param) => params[param]);
};

export const getTrueNumber = (amount, type = '', currency = 'â‚¬') => {
  const absAmount = Math.abs(amount);
  const strAmount = absAmount.toString();
  let sign = ' ';

  if (type !== '') {
    sign = type === 'credit' ? '+' : '-';
  }

  if (strAmount === '0') {
    return ` ${strAmount}${currency}`;
  }
  if (strAmount.length <= 2) {
    return `${sign}0,${strAmount}${currency}`;
  }

  const fistDigits = strAmount.substring(0, strAmount.length - 2);
  const lastDigits = strAmount.substring(strAmount.length - 2);

  if (lastDigits === '00') {
    return `${sign}${fistDigits}${currency}`;
  }
  return `${sign}${fistDigits},${lastDigits}${currency}`;
};
