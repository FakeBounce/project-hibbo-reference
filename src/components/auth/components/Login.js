import React from 'react';
import { Animated } from 'react-native';
import {
  shape,
  instanceOf,
  func,
  string,
  arrayOf,
  number,
  bool,
} from 'prop-types';
import KeyboardSpacer from 'react-native-keyboard-spacer';

import { getTranslations } from 'utils/i18n';

import Back from './Back';
import Input from './Input';
import Submit from './Submit';
import Link from './Link';
import FormErrors from './FormErrors';
import * as Styles from '../styles';

const Login = ({
  styling,
  errorApi,
  form,
  step,
  error,
  handleInputPosition,
  back,
  validate,
  onChange,
  inputPosition,
  errorMsg,
  changeAuthType,
  authProvider,
  editPhone,
  logginIn,
  inputTextSize,
}) => {
  return (
    <Styles.SContainerLogin style={styling}>
      <Back title="login" goBack={back} />

      <FormErrors
        error={error}
        errorMsg={errorMsg}
        inputPosition={inputPosition}
        errorApi={errorApi}
      />

      <Styles.SWrapperLogin
        onLayout={({ nativeEvent: { layout: { x, y } } }) => {
          handleInputPosition({ x, y });
        }}
      >
        <Input
          title="login"
          form={form.form}
          error={error}
          label={authProvider === '' ? form.label[step][0] : form.label[2][0]}
          name={authProvider === '' ? form.label[step][1] : form.label[2][1]}
          onChange={onChange}
          validate={authProvider === '' ? validate : editPhone}
          inputTextSize={inputTextSize}
        />
        <Styles.SBorderLogin />
      </Styles.SWrapperLogin>

      <Submit
        title="login"
        validate={authProvider === '' ? validate : editPhone}
        logginIn={logginIn}
      />

      {authProvider === '' && (
        <Styles.SLinksLogin>
          <Link
            label={getTranslations('auth.label.forgottenPwd')}
            action={() => {
              changeAuthType('resetPwd');
            }}
          />
          <Link
            label={getTranslations('auth.label.createAccount')}
            action={() => {
              changeAuthType();
            }}
          />
        </Styles.SLinksLogin>
      )}

      <KeyboardSpacer />
    </Styles.SContainerLogin>
  );
};

Login.propTypes = {
  logginIn: bool.isRequired,
  styling: shape({
    opacity: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  changeAuthType: func.isRequired,
  errorApi: string.isRequired,
  form: shape({
    form: shape({
      email: string.isRequired,
      password: string.isRequired,
    }).isRequired,
    label: arrayOf(arrayOf(string.isRequired)).isRequired,
  }).isRequired,
  step: number.isRequired,
  error: bool.isRequired,
  handleInputPosition: func.isRequired,
  back: func.isRequired,
  validate: func.isRequired,
  onChange: func.isRequired,
  inputPosition: shape({
    x: number.isRequired,
    y: number.isRequired,
  }).isRequired,
  errorMsg: string.isRequired,
  authProvider: string.isRequired,
  editPhone: func.isRequired,
  inputTextSize: number.isRequired,
};

export default Login;
