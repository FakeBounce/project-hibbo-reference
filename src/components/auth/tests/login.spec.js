import React from 'react';
import { Animated } from 'react-native';
import renderer from 'react-test-renderer';

import { getTranslations } from 'utils/i18n';

import Login from '../components/Login';

describe('Login', () => {
  const fn = jest.fn();
  const styling = {
    opacity: new Animated.Value(100),
  };
  const form = {
    label: [
      [getTranslations('auth.label.email'), 'email'],
      [getTranslations('auth.label.password'), 'password'],
    ],
    form: {
      email: '',
      password: '',
    },
  };
  const position = {
    x: 0,
    y: 0,
  };

  const component = renderer
    .create(
      <Login
        styling={styling}
        errorApi=""
        form={form}
        step={0}
        error={false}
        handleInputPosition={fn}
        back={fn}
        validate={fn}
        onChange={fn}
        errorMsg=""
        inputPosition={position}
        changeAuthType={fn}
      />,
    )
    .toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
