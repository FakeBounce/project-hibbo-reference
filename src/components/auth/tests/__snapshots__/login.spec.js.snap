// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the component 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "opacity": 100,
    }
  }
>
  <View
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={undefined}
  >
    <View
      style={undefined}
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        autoFocus={true}
        keyboardType="default"
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="Email"
        placeholderTextColor={undefined}
        returnKeyType="next"
        secureTextEntry={false}
        style={
          Array [
            undefined,
            Object {
              "color": undefined,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={undefined}
    />
  </View>
  <View
    style={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
        style={undefined}
      />
    </View>
  </View>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          Forgotten password
        </Text>
      </View>
    </View>
    <View
      style={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          I don't have an account
        </Text>
      </View>
    </View>
  </View>
</View>
`;
